### PR TITLE
Remove deprecated showDialog.child

### PR DESCRIPTION
**Pull request by Piinks**
## Description

This removes the deprecated `showDialog.child` parameter that has reached the EOL . The `builder` property should be used instead.

Part of deprecations that are slated for removal for next release in https://flutter.dev/go/deprecation-lifetime

More context: https://medium.com/flutter/deprecation-lifetime-in-flutter-e4d76ee738ad

## Related Issues

Deprecated in  https://github.com/flutter/flutter/pull/15303

## Tests

Tests for quick fix tool.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/fix_data.yaml [hunk 1]
```diff
@@ -11,6 +11,27 @@
...
+        argumentValue:
+          expression: '(context) => {% widget %}'
+          requiredIf: "widget != ''"
+          variables:
+            widget:
+              kind: fragment
+              value: 'arguments[child]'
+      - kind: 'removeParameter'
+        name: 'child'
+
```
Comment (Piinks) [inline packages/flutter/lib/fix_data.yaml:34]:
```suggestion

```

**Comment by Piinks** [comment]
@renyou is it safe to land this? I created a fix, and everything appears to pass. It isn't updating here though.

**Comment by renyou** [comment]
> @renyou is it safe to land this? I created a fix, and everything appears to pass. It isn't updating here though.

Looks like the something is stuck. Let me kick start a new test. Shouldn't take too long.

**Pull request merged**